std/node/assertion_error.ts

```diff
@@ -0,0 +1,582 @@
+}
+
+import { inspect } from "./util.ts";
+import { stripColor as removeColors } from "../fmt/colors.ts";
+
+// TODO: we should implement Node's concept of "primordials"
+// Ref: https://github.com/denoland/deno/issues/6040#issuecomment-637305828
+const MathMax = Math.max;
+const { Error } = globalThis;
+const {
+  create: ObjectCreate,
+  defineProperty: ObjectDefineProperty,
+  getPrototypeOf: ObjectGetPrototypeOf,
+  getOwnPropertyDescriptor: ObjectGetOwnPropertyDescriptor,
+  keys: ObjectKeys,
+} = Object;
+
+import { codes } from "./_errors.ts";
+const { ERR_INVALID_ARG_TYPE } = codes;
+
+let blue = "";
+let green = "";
+let red = "";
+let white = "";
+
+const kReadableOperator: { [key: string]: string } = {
+  deepStrictEqual: "Expected values to be strictly deep-equal:",
+  strictEqual: "Expected values to be strictly equal:",
+  strictEqualObject: 'Expected "actual" to be reference-equal to "expected":',
+  deepEqual: "Expected values to be loosely deep-equal:",
+  notDeepStrictEqual: 'Expected "actual" not to be strictly deep-equal to:',
+  notStrictEqual: 'Expected "actual" to be strictly unequal to:',
+  notStrictEqualObject:
+    'Expected "actual" not to be reference-equal to "expected":',
+  notDeepEqual: 'Expected "actual" not to be loosely deep-equal to:',
+  notIdentical: "Values have same structure but are not reference-equal:",
+  notDeepEqualUnequal: "Expected values not to be loosely deep-equal:",
+};
+
+// Comparing short primitives should just show === / !== instead of using the
+// diff.
+const kMaxShortLength = 12;
+
+export function copyError(source: Error): Error {
+  const keys = ObjectKeys(source);
+  const target = ObjectCreate(ObjectGetPrototypeOf(source));
+  for (const key of keys) {
+    const desc = ObjectGetOwnPropertyDescriptor(source, key);
+
+    if (desc !== undefined) {
+      ObjectDefineProperty(target, key, desc);
+    }
+  }
+  ObjectDefineProperty(target, "message", { value: source.message });
+  return target;
+}
+
+// TODO: create/find issue to suggest the following properties to be added to `Deno.inspect()` opts:
+// `customInspect`, `customInspect`, `showHidden`, `showProxy` and `getters` (and probably more)
+// Ref: https://nodejs.org/dist/latest-v14.x/docs/api/util.html#util_util_inspect_object_options
+export function inspectValue(val: unknown): string {
+  // The util.inspect default values could be changed. This makes sure the
+  // error messages contain the necessary information nevertheless.
+  return inspect(
+    val,
+    {
+      compact: false,
+      // customInspect: false,
+      depth: 1000,
+      // maxArrayLength: Infinity,
+      // Assert compares only enumerable properties (with a few exceptions).
+      // showHidden: false,
+      // Assert does not detect proxies currently.
+      // showProxy: false,
+      sorted: true,
+      // Inspect getters as we also check them when comparing entries.
+      // getters: true,
+    },
+  );
+}
```

Comment:
@nayeemrmn PTAL, does it make sense for `InspectOptions`?

Reply:
Resolved.